src/components/application_manager/src/event_engine/event_dispatcher_impl.cc

```diff
@@ -88,6 +88,14 @@ struct IdCheckFunctor {
   const unsigned long target_id;
 };
 
+void EventDispatcherImpl::remove_observer(const Event::EventID& event_id,
+                                          const int32_t hmi_correlation_id) {
+  auto& observers = observers_event_[event_id][hmi_correlation_id];
```

Comment:
@VladSemenyuk I think you should lock `state_lock_` before accessing `observers_event_` to avoid data races

Reply:
Done